src/java.base/linux/classes/sun/nio/fs/LinuxWatchService.java

```diff
@@ -70,8 +70,8 @@
             socketpair(sp);
             configureBlocking(sp[0], false);
         } catch (UnixException x) {
-            UnixNativeDispatcher.close(ifd);
-            throw new IOException(x.errorString());
+            throw UnixNativeDispatcher.close(ifd,
+                new IOException(x.errorString()), y -> y.asIOException(null));
```

Comment:
The most likely cause of socketpair failing here would be resources or out of file descriptors. When that happens then it's probably not interesting to also try to report the failure of inotify close. So I think this close can be close(ifd, e-> null) too.